### ISSUE TITLE
macro_gov (and related routines)

### ISSUE THREAD

**Issue by kslong** [sirocco-rt/sirocco#652]
Macro-gov is poorly commented and this needs to be improved, I have done a little (latest version of dev), but it would be better if someone who is more involved does this correctly..  
- Furthermore, it seems to return either 1 or 0, but this is not documented, nor is the reason for it identified. The returns are not used as far as I can tell.  
- Furthermore, it contains code that is impossible to get to, see below.  
- Additionally, it's not obvious, why one should be adding reverberation path lengths as is done below for the purposes of reverb.  macro_gov is not about moving photons from place to place.   - Also, the code is set up as a big while loop, but at various places we change the variable escape to 1 (which escapes the loop), and at other places we call return directly.    We should be consistent.   
- Finally the code operates very differently during spectral generation, but instead of dealing with this generically, a lot of code gets repeated (and I could imagine there is a more efficient way to do this, that would make the code more obvious during ionization cycles.)

Given our current issues with #648, we should clean this up.  

(There also smaller issues with the desciption of the routine macro_pops in the same file.  

```
          matom (p, nres, &escape);

          if (escape == 1)
          {
            /* It is going to escape as a r-packet that was created by de-activation of a macro atom.
               Therefore, if the frequency is suitable it should be recorded as a macro atom emission event for use in the
               computation of the k-packet emissivity needed for the final spectrum calculation. */
            *which_out = 1;

            //SWM - If reverb is on, and this is the last ionisation cycle, then track the photon path
            if (geo.reverb == REV_MATOM && geo.ioniz_or_extract && geo.fraction_converged > geo.reverb_fraction_converged)
            {
              line_paths_add_phot (&(wmain[p->grid]), p, nres);
            }
            return (1);
            /* 0803 - ksl - 60 - Added code to modify the photon origin to indicate the packet has been processed
               by a macro atom */
            if (p->origin < 10)
              p->origin += 10;
            return (1);
          }
        }

```

**Comment by kslong** [comment]
matom.c is another matom routine where the Doxygen commen ts do not follow the satandard approach.

**Comment by kslong** [comment]
estimators.c has some functions that have non-standard dox headers. This also needs fixing.

**Comment by jhmatthews** [comment]
Had a look a this, see pull #657 

Some comments:
* Commenting: looked pretty good to me, but I fixed a few places where doxygen format was missing and cleaned up a little bit of formatting.  
* Impossible code: I removed the first return in the above code snippet
* Returns: I made return always be 0 which in any case is not used. Perhaps the return statement is not needed in these cases because escape is set to 1, but I wasn't certain and didn't want to mess up such a critical routine. 
* Duplicate code / spectral cycles: I couldn't see a quick and easy way to avoid duplicating the geo.matom_radiation if statement. Possibly it could go in the larger if statements that identify the transition, but it needs to be done carefully so that the "fake_matom" routines are still entered in spectral cycles.
* Reverb: I think the reverb thing is probably correct. The path isn't being updated, but the photon is being flagged as one to track. But @smangham can confirm.
